File: cpp/25304.cpp
#include <iostream>

using namespace std;

int main() {
    int x, n;
    cin >> x >> n;

    for (int i = 0; i < n; ++i) {
        int a, b;
        cin >> a >> b;
        x -= a * b;
    }

    cout << (x == 0 ? "Yes" : "No");
    return 0;
}

File: python/25304.py
import sys

input = sys.stdin.readline

x = int(input())  # price on receipt
n = int(input())  # number of products

for _ in range(n):
    a, b = map(int, input().split())
    x -= a * b

print("Yes" if x == 0 else "No")
